Library/Homebrew/test/api/internal_tap_json/formula_spec.rb

```diff
@@ -0,0 +1,128 @@
+# frozen_string_literal: true
+
+describe "Internal Tap JSON -- Formula" do
+  let(:internal_tap_json) { File.read(TEST_FIXTURE_DIR/"internal_tap_json/homebrew-core.json").chomp }
+  let(:tap_git_head) { "9977471165641744a829d3e494fa563407503297" }
+
+  context "when generating JSON", :needs_macos do
+    before do
+      cp_r(TEST_FIXTURE_DIR/"internal_tap_json/homebrew-core", Tap::TAP_DIRECTORY/"homebrew")
+
+      # NOTE: Symlinks can't be copied recursively so we create them manually here.
+      (Tap::TAP_DIRECTORY/"homebrew/homebrew-core").tap do |core_tap|
+        mkdir(core_tap/"Aliases")
+        ln_s(core_tap/"Formula/f/fennel.rb", core_tap/"Aliases/fennel-lang")
+        ln_s(core_tap/"Formula/p/ponyc.rb", core_tap/"Aliases/ponyc-lang")
+      end
+    end
+
+    it "creates the expected hash" do
+      api_hash = CoreTap.instance.to_api_hash
+      api_hash["tap_git_head"] = tap_git_head # tricky to mock
+
+      expect(JSON.generate(api_hash)).to eq(internal_tap_json)
+    end
+  end
+
+  context "when loading JSON" do
+    before do
+      ENV["HOMEBREW_INTERNAL_JSON_V3"] = "1"
+      ENV.delete("HOMEBREW_NO_INSTALL_FROM_API")
+
+      allow(Homebrew::API).to receive(:fetch_json_api_file)
+        .with("internal/v3/homebrew-core.jws.json")
+        .and_return(JSON.parse(internal_tap_json), false)
+    end
+
+    it "loads tap aliases" do
+      expect(CoreTap.instance.alias_table).to eq({
+        "fennel-lang" => "fennel",
+        "ponyc-lang"  => "ponyc",
+      })
+    end
+
+    it "loads formula renames" do
+      expect(CoreTap.instance.formula_renames).to eq({
+        "advancemenu" => "advancemame",
+        "amtk"        => "libgedit-amtk",
+        "annie"       => "lux",
+        "antlr2"      => "antlr@2",
+        "romanesco"   => "fennel",
+      })
+    end
+
+    it "loads tap migrations" do
+      expect(CoreTap.instance.tap_migrations).to eq({
+        "adobe-air-sdk"          => "homebrew/cask",
+        "android-ndk"            => "homebrew/cask",
+        "android-platform-tools" => "homebrew/cask",
+        "android-sdk"            => "homebrew/cask",
+        "app-engine-go-32"       => "homebrew/cask/google-cloud-sdk",
+      })
+    end
+
+    it "loads tap git head" do
+      expect(Homebrew::API::Formula.tap_git_head)
+        .to eq(tap_git_head)
+    end
+
+    context "when loading formulae" do
+      let(:fennel_metadata) do
+        {
+          "dependencies"     => ["lua"],
+          "desc"             => "Lua Lisp Language",
+          "full_name"        => "fennel",
+          "homepage"         => "https://fennel-lang.org",
+          "installed"        => [],
+          "license"          => "MIT",
+          "name"             => "fennel",
+          "ruby_source_path" => "Formula/f/fennel.rb",
+          "tap"              => "homebrew/core",
+          "tap_git_head"     => tap_git_head,
+          "versions"         => { "bottle"=>true, "head"=>nil, "stable"=>"1.4.0" },
```

Comment:
```suggestion
          "dependencies"     => ["lua"],
          "desc"             => "Lua Lisp Language",
          "full_name"        => "fennel",
          "homepage"         => "https://fennel-lang.org",
          "license"          => "MIT",
          "name"             => "fennel",
          "ruby_source_path" => "Formula/f/fennel.rb",
          "tap"              => "homebrew/core",
          "tap_git_head"     => tap_git_head,
          "versions"         => { "bottle"=>true, "stable"=>"1.4.0" },
```

As these don't seem to be needed.

Do we use `desc` and `license` from the JSON?

Reply:
Yeah, I'll remove the installed array. That's just an oversight on my part. The fields here are not super important we're just trying to see that it loads somewhat correctly. It's a sanity check. Description and license are used by `brew info` and description is also used by `brew search` and `brew desc`.